5-Cookies/client-react/src/components/TodoList/TodoListItem/TodoListItem.driver.tsx

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import { render, fireEvent } from "@testing-library/react";
+import TodoListItem from "./TodoListItem";
+import { BaseDriver } from "../../../test/baseDriver";
+import dataHooks from "../../../dataHooks/dataHooks";
+import { mockTodo, throwErrorOnMissingWrapper } from "../../../test/utils";
+import { Item } from "../../../interfaces/interfaces";
+
+export class TodoListItemDriver extends BaseDriver {
+  private todoItem: Item = mockTodo();
+  private removeTodo: jest.Mock = jest.fn();
+
+  private buttonClickHandler = (dataHook: string) => {
+    if (!this.wrapper) return throwErrorOnMissingWrapper();
+    const buttonElem = this.wrapper.getByTestId(dataHook);
+    fireEvent.click(buttonElem);
+    return buttonElem;
+  };
+
+  private getTodoText = () => {
+    return this.wrapper?.getByText(this.todoItem.text).innerHTML;
+  };
+
+  private getTodoStatus = () => {
+    const switchElem = this.wrapper?.getByTestId(
+      dataHooks.todoToggleSwitch
+    ) as HTMLInputElement;
+    return switchElem.checked;
+  };
+
+  private getEditModeStatus = () => {
+    if (!this.wrapper) return throwErrorOnMissingWrapper();
+    try {
+      if (this.wrapper.getByTestId(dataHooks.editTodoItem)) return true;
+    } catch (error) {
+      return false;
+    }
```

Comment:
why `try..catch`? what are you expecting to catch here? if you're trying to check the existence of this element in the dom, there's probably a better way to do that

Reply:
This is what I'm trying to do and didn't find any better way to do it, do you have a suggestion on how to find the existence of an element?